### PR TITLE
Added immutable-js

### PR DESCRIPTION
**Pull request by herrmanno**
I added [immutable-js](https://github.com/facebook/immutable-js).
Note that I did not wrote the definition file mysel but used the one provided [here](https://github.com/facebook/immutable-js/blob/master/type-definitions/Immutable.d.ts).

I added the typical DefinitelyTyped header to the .d.ts file and set the original contributors  as 'Deifinitions by'. I hope that I did not infringe the original authors rights this way.

### CODE REVIEW

**Comment by vvakame** [comment]
*immutable/immutable.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/immutable
  * http://bower.io/search/?q=immutable
  * others?
* [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (immutable/immutable-tests.ts or others)
* [ ] pass the Travis-CI test?

File: immutable/LICENSE [hunk 1]
```diff
@@ -0,0 +1,30 @@
+BSD License
```
Comment (vvakame) [inline immutable/LICENSE]: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/LICENSE
this repository is MIT license.

**Comment by vvakame** [comment]
why this pull request be needed?

**Comment by herrmanno** [comment]
Because it would make it easier to create definitions for libraries that depends on immutable-js.

For example [flux](https://github.com/facebook/flux):

The MapStore class from flux/utils depends on immutable-js. If one would like to create a definition file for such a project, it would be helpful to just reference the dependency(immutable-js). When a user then installs the flux.d.ts file via tsd he could easily install immutable, too. In this case, the dependency to immutable.d.ts would be fulfilled, because the immutable.d.ts file would be at the right location.

**Comment by vvakame** [comment]
umm...
This repository is under MIT license.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/LICENSE

Can you got permission from the copyright owner? (@leebyron ?)

**Comment by leonyu** [comment]
I posted an issue on their GitHub. `immutable.js` is a very popular library, it might be beneficial to write one from scratch if licensing issue cannot be resolved. 

I have written `.d.ts.` for `omniscient` which has a hard dependency on `immutable.js`

**Comment by leebyron** [comment]
Im personally in favor of this but I'm mostly ignorant about how code sharing between differently licensed repos should work, but I'm sure we can clear this up. I'll talk to someone on my team who can speak to this and figure out what if anything we need to do.

**Comment by leebyron** [comment]
Ok, I checked this out with people who are knowledgeable about licenses and this PR is in great shape. 

The minimum requirement is an inclusion of the BSD license alongside the file, which this PR does, an alternative could be including the copy of the BSD license in a comment block in the file, but the separate file in the same directory is totally good. Including the PATENTS license is not required but totally acceptable. 

From my point of view (representing Facebook) this PR is totally fine from a licensing point of view.

**Comment by leonyu** [comment]
@leebyron Thank you for looking into this. Does the above also apply to `cursor.d.ts`?

**Comment by leebyron** [comment]
Yes it does

**Comment by leonyu** [comment]
@herrmanno do you mind preping this PR for merge? (update/squash/some basic tests) If not, I might have some time on my hand mid-January.

**Comment by polkovnikov-ph** [comment]
Is `cursor.d.ts` included? I don't see it in the list of files.

**Comment by polkovnikov-ph** [comment]
Any movement on this PR?

**Comment by leonyu** [comment]
I have started a new PR with `cursor.d.ts` included

**Comment by shovon** [comment]
While we wait for this PR to be merged, anyone wanting to import Immutable.js type definitions into their projects can do so by adding the following reference directive to their source code:

```typescript
/// <reference path="/path/to/node_modules/immutable/dist/immutable.d.ts" />
```

If you can't use npm for package management, you always have the option of downloading the type definitions directly from the Immutable.js repository, at [type-definitions/Immutable.d.ts](https://github.com/facebook/immutable-js/blob/master/type-definitions/Immutable.d.ts).

**Comment by RyanCavanaugh** [comment]
Hello and thank you for your contribution!

Due to an excessively long queue of pull requests that have become stale, we are "declaring bankruptcy" and closing all PRs opened before May 1, 2016. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

Going forward, we are committing to review or merge all PRs on a regular basis so this bankruptcy will not occur again. We apologize for the incovenience and hope you will continue to contribute to DefinitelyTyped in the future.

Thanks
The TypeScript and DefinitelyTyped teams

**Comment by polkovnikov-ph** [comment]
@RyanCavanaugh Are there licensing issues that need to be solved, or a new PR merged with master is needed? I'm totally lost with these policies and frustrated with TS infrastructure.

**Comment by leonyu** [comment]
Since TS 1.8, libraries can bundle their own typings, and it would be pull by TSC. That's another route to explore for `omniscient`/etc

**Comment by mhegazy** [comment]
> @RyanCavanaugh Are there licensing issues that need to be solved, or a new PR merged with master is needed? I'm totally lost with these policies and frustrated with TS infrastructure.

All the declaration files in this repo are provided under [one licence](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/LICENSE). Any contribution is expected to fall under this. We can not take contributions under other licences, and not doing so would put this repo, and its contributions under legal risk.

As noted by @leonyu, you are welcome to distribute your definition files using your own npm package. 

In the future, there should be support for redirects in the types publisher, see https://github.com/Microsoft/types-publisher/issues/4, which should allow you to redirect from definitely typed to any other repo you choose.


>  I'm totally lost with these policies and frustrated with TS infrastructure.

We do understand the frustration with type declaration acquisition. For this release cycle we have spent considerable energy to stream line this process (see https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/), and we do hope things are easier moving forward. We do appreciate your contribution and we sure want to make the contribution process as smooth as possible.

**Pull request closed without merging**